docs/linux/coverage.md

```diff
@@ -15,18 +15,112 @@ ln -s `which mips64el-linux-gnuabi64-readelf` ~/bin/mips64le/readelf
 export PATH=~/bin/mips64le:$PATH
 ```
 
-### objdump
+The target-triple prefix is determined based on the _target_ config option.
+
+### readelf
+
+`readelf` is used to detect virtual memory offset. Executor truncates PC values into `uint32` before sending them to `syz-manager` and `syz-manager` has to detect the offset.
 
-`objdump` is used to parse PC value of each call to `__sanitizer_cov_trace_pc` in the kernel image. These PC values are representing all code that is built into kernel image. PC values exported by kcov are compared against these to determine coverage.
+
+## Reporting coverage data
+
+_MakeReportGenerator_ factory creates an object database for the report. It requires target data, as well as information on the location of the source files and build directory. The first step in building this database is 
+extracting the function data from the target binary.
 
 ### nm
+`nm` is used to parse address and size of each function in the kernel image
+```
```

Comment:
Empty lines around \`\`\`.
` bash` after \`\`\` for shell snippets.
Here and below.